webapp/src/components/testing/Testing.vue

```diff
@@ -288,4 +295,162 @@ function taskTitle(taskID: string): string | undefined {
+  },
+  {
+    element: ".mb-1:nth-child(1)",
+    popover: {
+      description:
+        "Run the model on a validation dataset to assess its accuracy.",
+      side: "top",
+      align: "center",
+    },
+  },
+  {
+    element: ".mb-1:nth-child(2)",
+    popover: {
+      description:
+        "Use the selected model to make predictions on new data inputs.",
+      side: "top",
+      align: "center",
+    },
+  },
+  {
+    element: "#model-repository",
+    popover: {
+      title: "DISCO Model Repository",
+      description:
+        "Download additional models from the repository for testing or federated training tasks.",
+      side: "top",
+      align: "center",
+    },
+  },
+];
+// Define the steps for the guide related to data handling
+const dataSteps: DriveStep[] = [
+  {
+    popover: {
+      title: "Expected Data Format",
+      description:
+        "Testing your model on unseen data is crucial. Make sure the dataset you select was not part of the training process to ensure accurate evaluation.",
+      side: "top",
+      align: "center",
+    },
+  },
+  {
+    element: "#data-link-download",
+    popover: {
+      title: "Test",
+      description:
+        "Click here to download a sample dataset if you don't have your own. Perfect for testing and experimenting!",
+      side: "top",
+      align: "center",
+    },
+  },
+  {
+    element: "#select-data-bttn",
+    popover: {
+      title: "Upload Your Dataset",
+      description:
+        "Click here to upload your dataset for model training. Your data stays secure and never leaves your device.",
+      side: "top",
+      align: "center",
+    },
+  },
+];
+// Define the steps for the guide related to model evaluation
+const evaluationSteps: DriveStep[] = [
+  {
+    popover: {
+      title: "Model Evaluation",
+      description:
+        "This page is intended for testing or validating the selected model. After the testing is complete, the results can be saved in CSV file format for further analysis or use.",
+      side: "top",
+      align: "center",
+    },
+  },
+];
+watch(
+  () => guideStoreTesting.showGuide,
+  (newValue) => {
+    if (newValue) {
+      if (validationStore.step === 0) {
+        driverTesting.destroy();
+        if (models.infos.isEmpty()) {
+          driverTesting.setSteps(modelEmptySteps);
+        } else {
+          driverTesting.setSteps(modelSteps);
+        }
+        driverTesting.drive();
+        guideStoreTesting.toggleGuide();
+      }
+      if (validationStore.step !== 0) {
+        if (validationStore.step === 1) {
+          driverTesting.destroy();
+          driverTesting.setSteps(dataSteps);
+          driverTesting.drive();
+          guideStoreTesting.toggleGuide();
+        }
+        if (validationStore.step === 2) {
+          driverTesting.destroy();
+          driverTesting.setSteps(evaluationSteps);
+          driverTesting.drive();
+          guideStoreTesting.toggleGuide();
+        }
+      }
+    }
+  },
+);
```

Comment:
many path are similar and can be commonized


also, you can avoid the not-really-reactive call to watch by [destructuring the store](https://pinia.vuejs.org/core-concepts/#Destructuring-from-a-Store):
```ts
const { showGuide } = storeToRefs(useGuideStore());
```

```suggestion
watch(showGuide, (showGuide) => {
  if (!showGuide) return;

  driverTesting.destroy();
  switch (validationStore.step) {
    case 0:
      if (models.infos.isEmpty()) {
        driverTesting.setSteps(modelEmptySteps);
      } else {
        driverTesting.setSteps(modelSteps);
      }
      break;
    case 1:
      driverTesting.setSteps(dataSteps);
      break;
    case 2:
      driverTesting.setSteps(evaluationSteps);
      break;
  }
  driverTesting.drive();
  showGuide.value = false;
});
```